src/hooks/serverStateHooks/useGetFollowerList.ts

```diff
@@ -0,0 +1,24 @@
+import { useQuery } from '@tanstack/react-query';
+import { useSearchParams } from 'next/navigation';
+
+import getFollowerList from '@/apis/follow/getFollowerList';
+import { USER } from '@/constants/API';
+import { FollowUser } from '@/types/user';
+
+const useGetFollowerList = () => {
+  const searchParams = useSearchParams();
+  const userIdParam = searchParams.get('userId');
+  const userId = userIdParam ? Number(userIdParam) : null;
+
+  const { data, isLoading } = useQuery({
+    queryKey: [USER.followerList(userId as number)],
+    queryFn: () => getFollowerList(userId as number),
+  });
+
+  return {
+    followerList: data !== undefined ? data.users : ([] as FollowUser[]),
```

Comment:
```suggestion
    followerList: data?.users ?? [],
```

`data`가 없으면 빈 배열을 반환하도록 작성할 수도 있을 것 같아요!